src/batch/src/executor/join/hash_join.rs

```diff
@@ -2455,16 +2464,38 @@ mod tests {
         }
     }
 
-    fn is_data_chunk_eq(left: &DataChunk, right: &DataChunk) -> bool {
+    fn is_data_chunk_bag_eq(left: &DataChunk, right: &DataChunk) -> bool {
         assert!(left.is_compacted());
         assert!(right.is_compacted());
 
         if left.cardinality() != right.cardinality() {
             return false;
         }
 
-        left.rows()
-            .zip_eq_debug(right.rows())
+        // Sort and compare
+        let column_orders = (0..left.columns().len())
+            .map(|i| ColumnOrder::new(i, OrderType::ascending()))
+            .collect_vec();
+        let left_encoded_chunk = encode_chunk(left, &column_orders).unwrap();
+        let mut sorted_left = left_encoded_chunk
+            .into_iter()
+            .enumerate()
+            .map(|(row_id, row)| (left.row_at_unchecked_vis(row_id), row))
+            .collect_vec();
+        sorted_left.sort_unstable_by(|(_, a), (_, b)| a.cmp(b));
+
+        let right_encoded_chunk = encode_chunk(right, &column_orders).unwrap();
+        let mut sorted_right = right_encoded_chunk
+            .into_iter()
+            .enumerate()
+            .map(|(row_id, row)| (right.row_at_unchecked_vis(row_id), row))
+            .collect_vec();
+        sorted_right.sort_unstable_by(|(_, a), (_, b)| a.cmp(b));
+
+        sorted_left
+            .into_iter()
+            .map(|(row, _)| row)
+            .zip_eq_debug(sorted_right.into_iter().map(|(row, _)| row))
```

Comment:
Should we put this in a separate function? Not sure if it will break the semantics of some tests. Previously we expect the unsorted ordering to be equal. How come we change it now 🤔